paddle/framework/init_test.cc

```diff
@@ -20,7 +20,26 @@ TEST(InitDevices, CPU) {
   using paddle::framework::InitDevices;
   using paddle::platform::DeviceContextPool;
 
+#ifndef PADDLE_WITH_CUDA
   InitDevices();
   DeviceContextPool& pool = DeviceContextPool::Instance();
-  ASSERT_GE(pool.size(), 1U);
+  ASSERT_EQ(pool.size(), 1U);
+#endif
+}
+
+TEST(InitDevices, CUDA) {
+  using paddle::framework::InitDevices;
+  using paddle::platform::DeviceContextPool;
+
+  int count = 0;
+  try {
+    count = paddle::platform::GetCUDADeviceCount();
+  } catch (const std::exception& exp) {
+  }
```

Comment:
If `#ifdef PADDLE_WITH_CUDA` is placed in line 31, I think `try ... catch` is unnecessary.

Reply:
done